app/services/anno_auth_validator.rb

```diff
@@ -0,0 +1,43 @@
+require 'json'
+require 'net/http'
+require 'jwt'
+
+class AnnoAuthValidator
+
+    def initialize(authHeader, canvas)
+        @authHeader = authHeader
+        @canvas = canvas
+      end
+
+    # Verfiying from remote destination that the token sent has the same group ID for the canvas and that the user has access
+    def authorize
+        begin
+        token = decodeToken(@authHeader)
+  
+         data = getCanvasData(@canvas, token.first["user_id"])
+  
+        if data["group_id"] == token.first["group_id"] && data["has_access"] == "true"
+            return true
+        else 
+            return false
+        end
+        rescue Exception
+            return false
+        end
+    end
+
+    def decodeToken authHeader
+        auth = authHeader.split(' ').last
+        JWT.decode auth, Rails.application.config.JWT_Password
+    end
+
+    def getCanvasData canvas, user_id
+        url = Rails.application.config.JWT_Canvas_Verification_Url + "?canvas_id=" + canvas + "&user_id=" + user_id
+        uri = URI(url)
+        res = Net::HTTP.get(uri)
+        JSON.parse(res).first
+    end
+
+    private :decodeToken, :getCanvasData
```

Comment:
I think we generally prefer the less specific way of setting methods as private:

```ruby
class Foo
  def public_method; end

  private

  def secret_method; end
end
```